contrib/babelfishpg_tsql/src/pltsql_utils.c

```diff
@@ -2164,7 +2188,16 @@ static List
 				appendStringInfo(&query, "GRANT dummy ON ALL TABLES IN SCHEMA dummy TO dummy WITH GRANT OPTION; ");
 			else
 				appendStringInfo(&query, "GRANT dummy ON ALL TABLES IN SCHEMA dummy TO dummy; ");
-			appendStringInfo(&query, "ALTER DEFAULT PRIVILEGES IN SCHEMA dummy GRANT dummy ON TABLES TO dummy; ");
+			
+			if (owner_other_than_dbo)
+			{
+				/* Grant ALTER DEFAULT PRIVILEGES on schema owner and dbo user. */
+				appendStringInfo(&query, "ALTER DEFAULT PRIVILEGES FOR ROLE dummy IN SCHEMA dummy GRANT dummy ON TABLES TO dummy; ");
+				appendStringInfo(&query, "ALTER DEFAULT PRIVILEGES FOR ROLE dummy IN SCHEMA dummy GRANT dummy ON TABLES TO dummy; ");
+				expected_stmts = 3;
+			}
+			else
+				appendStringInfo(&query, "ALTER DEFAULT PRIVILEGES IN SCHEMA dummy GRANT dummy ON TABLES TO dummy; ");
```

Comment:
Is there a possibility of db_owner role as an owner of any schema -- dbo schema of master, tempdb and msdb databases have db_owner roles as an owner?
If that's the case then for db_owner also we will need to execute ALTER DEF. PRIVILEGES for db_owner.

Reply:
Do we need this now? Or any test case which you think should be added for this? 
There are existing tests for schema which are either owned by `dbo` or `db_owner`. 
```
ALTER DEFAULT PRIVILEGES IN SCHEMA dummy GRANT dummy ON TABLES TO dummy;
```
This statement seems to handle those scenarios because only the owners of schema could execute the GRANT statement and this statement will take the role as current user. That's why I didn't specify "dbo" here.